pkg/validations/extendedversion.go

```diff
@@ -31,3 +51,60 @@ func validateBundleSignature(bundle *v1alpha1.Bundles) error {
 	}
 	return nil
 }
+
+func isExtendedSupport(clusterSpec *anywherev1.Cluster, bundle *v1alpha1.Bundles) (bool, error) {
+	versionsBundle, err := cluster.GetVersionsBundle(clusterSpec.Spec.KubernetesVersion, bundle)
+	if err != nil {
+		return false, fmt.Errorf("getting versions bundle for %s kubernetes version: %w", clusterSpec.Spec.KubernetesVersion, err)
+	}
+
+	endOfStandardSupport, err := parseDate("2006-01-02", versionsBundle.EndOfStandardSupport)
+	if err != nil {
+		return false, fmt.Errorf("parsing EndOfStandardSupport field format: %w", err)
+	}
+
+	return isPastDateThanToday(endOfStandardSupport), nil
+}
+
+func getLicense(clusterSpec *anywherev1.Cluster) (*jwt.Token, error) {
+	if clusterSpec.Spec.LicenseToken == "" {
+		return nil, errors.New("license token is required for extended kubernetes support")
+	}
+	token, err := signature.ParseLicense(clusterSpec.Spec.LicenseToken, constants.LincesePublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("parsing license: %w", err)
+	}
+
+	return token, nil
+}
+
+func validateLicense(token *jwt.Token) error {
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return errors.New("could not parse the licenseKey claims")
+	}
+	endValidity, ok := claims["endValidity"].(string)
+	if !ok {
+		return errors.New("license validity field missing, not a valid license")
+	}
+
+	validity, err := parseDate(time.RFC3339, endValidity)
+	if err != nil {
+		return fmt.Errorf("parsing endValidity field from licenseKey: %w", err)
+	}
+
+	if isPastDateThanToday(validity) {
+		return errors.New("license is expired, please renew the license")
+	}
+
+	return nil
+}
+
+func isPastDateThanToday(dateToCompare time.Time) bool {
+	today := time.Now().Truncate(24 * time.Hour)
+	return dateToCompare.Before(today)
+}
+
+func parseDate(format string, date string) (time.Time, error) {
```

Comment:
is this function really needed? it takes the exact same inputs and returns the exact same outputs as `time.Parse`?

Reply:
`versionsBundle.EndOfStandardSupport` is in the format `yyyy-mm-dd` where as `endValidity` in the jwt is in the format `yyyy-mm-ddThh:mm:ss.000Z`. Thus I created a function which returns date based on the format we specify.